profile-widget/profilewidget2.cpp

```diff
@@ -1095,6 +1095,38 @@ void ProfileWidget2::setProfileState(const dive *dIn, int dcIn)
 	setProfileState();
 }
 
+// Update visibility of non-interactive chart features according to preferences
+void ProfileWidget2::updateVisibility()
+{
+#ifndef SUBSURFACE_MOBILE
+	pn2GasItem->setVisible(prefs.pp_graphs.pn2);
+	po2GasItem->setVisible(prefs.pp_graphs.po2);
+	pheGasItem->setVisible(prefs.pp_graphs.phe);
+
+	const struct divecomputer *currentdc = d ? get_dive_dc_const(d, dc) : nullptr;
+	bool setpointflag = currentdc && currentdc->divemode == CCR && prefs.pp_graphs.po2;
+	bool sensorflag = setpointflag && prefs.show_ccr_sensors;
+	o2SetpointGasItem->setVisible(setpointflag && prefs.show_ccr_setpoint);
+	ccrsensor1GasItem->setVisible(sensorflag);
+	ccrsensor2GasItem->setVisible(currentdc && sensorflag && currentdc->no_o2sensors > 1);
+	ccrsensor3GasItem->setVisible(currentdc && sensorflag && currentdc->no_o2sensors > 2);
+	ocpo2GasItem->setVisible(currentdc && currentdc->divemode == PSCR && prefs.show_scr_ocpo2);
+
+	heartBeatItem->setVisible(prefs.hrgraph);
+#endif
```

Comment:
I realize that you are just moving code... but having just one `#ifdef` block would be easier to read, don't you think?

Also (and completely unrelated) there's a typo in the commit message of **profile: factor out updating of visibility from setProfileState()** "factor out" instead of "factor our"

Reply:
Yes, but note that these ifdefs will be removed later anyway.

Basically, I think the whole idea of mobile vs. desktop profiles is broken. This should rather depend on the size of the profile. For example on big tablets you might want to show all features. But that is for the future...